src/Type/Php/ClosureFromCallableDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,48 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\Native\NativeParameterReflection;
+use PHPStan\Type\ClosureType;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+
+class ClosureFromCallableDynamicReturnTypeExtension implements \PHPStan\Type\DynamicStaticMethodReturnTypeExtension
+{
+
+	public function getClass(): string
+	{
+		return \Closure::class;
+	}
+
+	public function isStaticMethodSupported(MethodReflection $methodReflection): bool
+	{
+		return $methodReflection->getName() === 'fromCallable';
+	}
+
+	public function getTypeFromStaticMethodCall(MethodReflection $methodReflection, StaticCall $methodCall, Scope $scope): Type
+	{
+		$callableType = $scope->getType($methodCall->args[0]->value);
+		if ($callableType->isCallable()->no()) {
+			return new ErrorType();
+		}
+
+		$closureTypes = [];
+		foreach ($callableType->getCallableParametersAcceptors($scope) as $variant) {
+			/** @var NativeParameterReflection[] $parameters */
```

Comment:
I'm not too fond about this, it's there just to stop phpstan complaining. If I understand correctly why it's there, maybe it's okay to lift this restriction in ClosureType now?

Reply:
Yes, you can try making ClosureType constructor more general. In a new commit please, where this phpDoc is also removed.